src/components/DateInput/DateInput.vue

```diff
@@ -6,28 +6,42 @@
       :label="label"
       :sr-only-label="srOnlyLabel"
       :size="size"
+      :error="error"
       @click.stop="isOpen = !isOpen"
       @keydown.space.stop="isOpen = !isOpen"
-    />
+    >
+      <template #iconRight>
+        <Calendar class="w-5 h-5 text-gray-900" />
+      </template>
+    </TextInput>
     <Datepicker
       :id="datepickerId"
       v-model="date"
       v-model:open="isOpen"
       :min="min"
       :max="max"
       :class="['absolute w-72', {'right-0' : rightJustified}]"
+      :disable-weekends="disableWeekends"
     />
+    <div
+      v-if="error && errorMessage"
+      class="mt-2 ml-1 text-error text-xs flex items-center"
+    >
+      <AlertCircle class="w-4 h-4 mr-1" /> {{ errorMessage }}
+    </div>
```

Comment:
Error messages need to be properly associated with the input to make them accessible.  This needs to be defined as a live region so the error is announced to screen reader users (more on how to do that here: https://egghead.io/lessons/react-define-a-live-region-to-ensure-dynamic-changes-are-announced-by-assistive-technologies).  Additionally I think we probably should add an `aria-describedby` dynamically (only when the error exists) to the `<input>` that points to an `id` on this `div`

Reply:
This is looking great, just one more thing.  For a live region to work it needs to always be in the DOM.  So we can't have the `v-if="error && errorMessage"` dynamically adding it later.  We need to structure this so that the element is in the DOM but maybe stylistically hidden and empty until there is an error.  That allows the live region to detect when there is a change (i.e. when the text gets dynamically added) and that's when it will read the change to the user.

Reply:
Also we probably want to use `aria-live="polite"` instead of `aria-live="assertive"` so that the screen reader doesn't end up interrupting itself if there ends up being multiple live regions on the page, since this is a shared component and we can't know all of the places it will be used.

Reply:
thanks! I'll try it with `polite` and `v-show`

Reply:
[v-show](https://vuejs.org/guide/essentials/conditional.html#v-show)
> The difference is that an element with v-show will always be rendered and remain in the DOM; v-show only toggles the display CSS property of the element.

Reply:
I think you may want to add `{{ error && errorMessage}}` so that the text of the error message is not in the DOM until there is actually an error.  This will make sure the screen reader doesn't read the message earlier than it should.  Even though it's visually hidden doesn't mean it will be hidden from the screen reader.

Reply:
hm, my idea was that `error : false`, and `errorMessage: null` by default, and they would only be set and/or cleared at the same time by the parent
are you saying this is like a just-in-case fallback?

Reply:
also this looks like react syntax, I didn't know we can do it like that on vue! (can we?)

Reply:
> or that we should be able to set the errorMessage that'll always be the same, and not have to set it and clear it along with the error boolean

Correct.  We don't always know how the parent is going to use this component.  They may use a static `errorMessage` and then just change `error` when there is an error.  So this should be able to work with either approach.  In fact, I would be more likely to assume `errorMessage` doesn't change but `error` does.

> also this looks like react syntax
Nope, not React syntax, just JavaScript syntax.  It's essentially a conditional shortcut that says: if the first condition is `true` return the second, else skip.